### ISSUE TITLE
Reproducing Results from Paper

### ISSUE THREAD

**Issue by xboldt** [gregorbachmann/Next-Token-Failures#3]
Hello,  

Thank you for your work on this paper! I’ve been trying to reproduce the results but haven’t been able to achieve the expected nearly 100% accuracy on (G_{2,5}) and (G_{2,3}). I’d appreciate any guidance on what might be missing in my setup.  

#### What I Tried  
I have trained the following models from scratch:  
- GPT-mini (using standard, teacherless, and reverse settings)  

and fine-tuned the following ones:
- GPT-2  
- GPT-2 Large  

I followed the hyperparameters mentioned in the paper:  

```bash
python train.py --model gpt --n_layer 12 --n_embd 384 --n_head 6 --n_train 200000 --n_test 20000  --batch_size 256 --dataset graph --deg 2 --path 5 --num_nodes 50 --lr 0.0005 --epochs 500
```

However, none of these configurations led to the near-100% accuracy reported for (G_{2,5}) (and (G_{2,3})), actually I am getting accuracies below 1% for any of the settings. Is the code in this repo the code which produced the results in the paper? Could you provide the actual hyperparameter settings you used to train your models?

Thanks in advance for your help!

**Comment by xboldt** [comment]
I fixed the issue by turning off FlashAttention. It seems that FlashAttention had some compatibility issues with my current version of PyTorch. After turning off use_flash, the results are consistent, and everything works as expected now.

**Comment by gregorbachmann** [comment]
Hi Sylwester, 

Thanks for your interest! Glad to hear you could fix things quickly :) 

Let me know if I can help with anything else!

Best,
Gregor
